src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -111,6 +113,7 @@ public Pipe(PipeOptions options)
             // If we're using the default pool then mark it as null since we're just going to use the
             // array pool under the covers
             _pool = options.Pool == MemoryPool<byte>.Shared ? null : options.Pool;
+            _maxPooledBufferSize = _pool?.MaxBufferSize ?? 0;
```

Comment:
This is an interface call, so pushing it up to constructor rather than calling it twice per `AllocateSegment`